acceptance-tests/src/test/java/it/com/atlassian/bitbucket/jenkins/internal/SmokeTest.java

```diff
@@ -0,0 +1,206 @@
+import it.com.atlassian.bitbucket.jenkins.internal.pageobjects.BitbucketTokenCredential;
+import it.com.atlassian.bitbucket.jenkins.internal.pageobjects.BitbucketWebhookTrigger;
+import it.com.atlassian.bitbucket.jenkins.internal.util.BitbucketUtils.*;
+import org.apache.commons.io.FileUtils;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.lib.TextProgressMonitor;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.eclipse.jgit.transport.CredentialsProvider;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeDiagnosingMatcher;
+import org.jenkinsci.test.acceptance.junit.AbstractJUnitTest;
+import org.jenkinsci.test.acceptance.junit.WithPlugins;
+import org.jenkinsci.test.acceptance.plugins.credentials.CredentialsPage;
+import org.jenkinsci.test.acceptance.plugins.credentials.UserPwdCredential;
+import org.jenkinsci.test.acceptance.po.Build;
+import org.jenkinsci.test.acceptance.po.FreeStyleJob;
+import org.jenkinsci.test.acceptance.po.JenkinsConfig;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static io.restassured.http.ContentType.JSON;
+import static it.com.atlassian.bitbucket.jenkins.internal.util.BitbucketUtils.*;
+import static java.time.Duration.ofMinutes;
+import static org.apache.commons.lang3.RandomStringUtils.randomNumeric;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.jenkinsci.test.acceptance.plugins.credentials.ManagedCredentials.DEFAULT_DOMAIN;
+import static org.jenkinsci.test.acceptance.po.Build.Result.SUCCESS;
+
+@WithPlugins({"mailer", "atlassian-bitbucket-server-integration"})
+public class SmokeTest extends AbstractJUnitTest {
+
+    @Rule
+    public final TemporaryFolder tempFolder = new TemporaryFolder();
+
+    private String bbsAdminCredsId;
+    private BitbucketRepository forkRepo;
+    private String serverId;
+
+    @Before
+    public void setUp() {
+        // BB Access Token
+        PersonalToken bbsToken = createPersonalToken(PROJECT_READ_PERMISSION, REPO_ADMIN_PERMISSION);
+        CredentialsPage credentials = new CredentialsPage(jenkins, DEFAULT_DOMAIN);
+        credentials.open();
+        BitbucketTokenCredential bbsTokenCreds = credentials.add(BitbucketTokenCredential.class);
+        bbsTokenCreds.token.set(bbsToken.getSecret());
+        bbsTokenCreds.description.sendKeys(bbsToken.getId());
+        credentials.create();
+
+        // BB Admin user/password
+        credentials.open();
+        UserPwdCredential bbAdminCreds = credentials.add(UserPwdCredential.class);
+        bbsAdminCredsId = "admin-" + randomNumeric(8);
```

Comment:
In hindsight, I probably should use `UUID`s instead of the weaker `RandomStringUtils.randomNumeric()` :thinking: